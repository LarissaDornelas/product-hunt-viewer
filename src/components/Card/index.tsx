import React from "react";

import { Container } from "./styles";

const Card: React.FC = (props) => {
  const { children } = props;
  return <Container data-testid="card-container">{children}</Container>;
};

export { Card };
