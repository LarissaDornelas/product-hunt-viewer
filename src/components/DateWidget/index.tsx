import React from "react";
import { format } from "date-fns";

import { Container } from "./styles";

const DateWidget: React.FC = () => {
  return (
    <Container data-testid="date-widget-container">
      <p>{format(new Date(), "'Today', Lo LLL ")}</p>
    </Container>
  );
};

export { DateWidget };
