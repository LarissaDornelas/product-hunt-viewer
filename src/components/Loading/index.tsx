import React from "react";
import { Container } from "./styles";

const Loading: React.FC = () => {
  return <Container data-testid="loading-container" />;
};

export { Loading };
