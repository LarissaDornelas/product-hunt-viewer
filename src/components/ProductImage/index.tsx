import React from "react";

import { Container } from "./styles";
import { IProductImage } from "./types";

const ProductImage: React.FC<IProductImage> = (props) => {
  const { imageUrl, size } = props;

  return (
    <Container
      data-testid="product-image-container"
      imageUrl={imageUrl}
      size={size}
    />
  );
};

export { ProductImage };
