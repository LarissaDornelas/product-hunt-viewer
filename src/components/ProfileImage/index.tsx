import React from "react";

import { Container } from "./styles";
import { IProfileImage } from "./types";

const ProfileImage: React.FC<IProfileImage> = (props) => {
  const { imageUrl } = props;

  return (
    <Container data-testid="profile-image-container" imageUrl={imageUrl} />
  );
};

export { ProfileImage };
