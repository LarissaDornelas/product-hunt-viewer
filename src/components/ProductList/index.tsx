import React, { useRef, useEffect, useCallback } from "react";

import { Container, LoadingContainer, ObservableContainer } from "./styles";
import { IProductItem, IProductList } from "./types";
import { ProductSummary } from "components";
import { Loading } from "components/Loading";

const ProductList: React.FC<IProductList> = (props) => {
  const { data, handleFetchMore, loading } = props;

  const scrollObserver = useRef() as React.MutableRefObject<any>;

  const loadMore = useCallback(
    (entries) => {
      const target = entries[0];

      if (target.isIntersecting) {
        handleFetchMore();
      }
    },
    [handleFetchMore]
  );

  useEffect(() => {
    const observer = new IntersectionObserver(loadMore);

    if (scrollObserver && scrollObserver.current) {
      observer.observe(scrollObserver.current);
    }

    return () => observer.disconnect();
  }, [scrollObserver, loadMore]);

  return (
    <Container>
      {data.length > 0 && (
        <>
          {data.map((item: IProductItem, index: number) => (
            <ProductSummary key={index} {...item} />
          ))}
          {loading && (
            <LoadingContainer>
              <Loading />
            </LoadingContainer>
          )}
          <ObservableContainer ref={scrollObserver}></ObservableContainer>
        </>
      )}
    </Container>
  );
};

export { ProductList };
