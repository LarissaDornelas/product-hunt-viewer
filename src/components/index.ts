import { ProfileImage } from "./ProfileImage";
import { DateWidget } from "./DateWidget";
import { SearchButton } from "./SearchButton";
import { Tab } from "./Tab";
import { Card } from "./Card";

export { ProfileImage, DateWidget, SearchButton, Tab, Card };
