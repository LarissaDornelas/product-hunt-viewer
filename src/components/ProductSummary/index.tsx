import React from "react";

import { Card, ProductImage, VoteButton } from "components";
import { ImageSize } from "components/ProductImage/types";
import { Container, DataContainer } from "./styles";

import { IProductItem } from "components/ProductList/types";

const ProductSummary: React.FC<IProductItem> = (props) => {
  const { name, description, thumbnail, votesCount } = props;

  return (
    <Card>
      <Container>
        <ProductImage imageUrl={thumbnail} size={ImageSize.MEDIUM} />
        <DataContainer>
          <h1>{name}</h1>
          <p>{description}</p>
        </DataContainer>
        <VoteButton votes={+votesCount} />
      </Container>
    </Card>
  );
};

export { ProductSummary };
