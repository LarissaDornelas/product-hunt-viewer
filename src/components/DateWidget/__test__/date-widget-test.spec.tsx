import React from "react";
import ReactDOM from "react-dom";
import { format } from "date-fns";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import { DateWidget } from "components";

describe("Component: DateWidget", () => {
  afterEach(cleanup);
  const todayDate = format(new Date(), "'Today', Lo LLL ");

  it("Should render without crashing", () => {
    const container = document.createElement("div");
    ReactDOM.render(
      <DateWidget>
        <p></p>
      </DateWidget>,
      container
    );
  });

  it("Should render DateWidget correctly", () => {
    const { getByTestId } = render(
      <DateWidget>
        <p>{todayDate}</p>
      </DateWidget>
    );

    expect(getByTestId("date-widget-container")).toContainHTML(
      `<p>${todayDate}</p>`
    );
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <DateWidget>
          <p>{todayDate}</p>
        </DateWidget>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
