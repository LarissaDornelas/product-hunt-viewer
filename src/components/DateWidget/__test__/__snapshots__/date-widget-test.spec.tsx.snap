// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DateWidget matches snapshot 1`] = `
<div
  className="sc-gsTCUz cKpBUT"
  data-testid="date-widget-container"
>
  <p>
    Today, 3rd Mar 
  </p>
</div>
`;
